components/sup/src/manager/service.rs

```diff
@@ -921,11 +934,23 @@ impl Service {
         win_perm::harden_path(path.as_ref())
     }
 
+    /// Called when the Supervisor reattaches itself to an already
+    /// running service. Use this to re-initiate any associated
+    /// processes, futures, etc.
+    ///
+    /// This should generally be the opposite of `Service::detatch`.
+    fn reattach(&mut self, executor: &TaskExecutor) {
+        outputln!("Reattaching to {}", self.service_group);
+        self.initialized = true;
```

Comment:
This one's a bit surprising since `detach` doesn't set it to false. I'm assuming that's because after `detach` is run, we're shutting down the whole supervisor, so this will only get called when it's false again.

This tends to indicate to me that a `bool` is no longer sufficiently expressive for what we're trying to model, and we may be better served by an `enum`.

Reply:
Agreed... I'm going to wait a little bit to dive into a larger refactor, though, because it appears that it will have a bigger impact than I was initially anticipating.